src/environment.rs

```diff
@@ -143,9 +134,41 @@ impl Environment {
         RoTransaction::new(self)
     }
 
+    fn float_to_usize(val: f64) -> usize {
+        if val > usize::MAX as f64 {
+            panic!("Failed to convert headroom value to usize (MAX hit); this means either database configuration is wrong or an invariant is broken");
+        } else if val < usize::MIN as f64 {
+            panic!("Failed to convert headroom value to usize (MIN hit); this means either database configuration is wrong or an invariant is broken");
+        }
```

Comment:
Maybe NaN shoudl be handled too...

Reply:
Yeah, this is my fear... there's always some corner case. Damn. I'll see how to handle this.

Reply:
Alrighty. No more floating point numbers here. I replaced this with an integer.